src/buildHTML.js

```diff
@@ -71,23 +103,25 @@ export const buildExpression = function(expression, options, isRealGroup,
 
     // Ignore explicit spaces (e.g., \;, \,) when determining what implicit
     // spacing should go between atoms of different classes, and add dummy
-    // spans for determining spacings between surrounding atoms
-    const nonSpaces = [
-        surrounding[0] && makeSpan([surrounding[0]], [], options),
+    // spans for determining spacings between surrounding atoms.
+    const nonSpaces: (?HtmlDomNode)[] = [
+        surrounding[0] ? makeSpan([surrounding[0]], [], options) : null,
         ...rawGroups.filter(group => group && group.classes[0] !== "mspace"),
-        surrounding[1] && makeSpan([surrounding[1]], [], options),
+        surrounding[1] ? makeSpan([surrounding[1]], [], options) : null,
```

Comment:
No action necessary here.  Currently we only know the atom type for a group until after we build the HTML for it.  Eventually, it would be nice if we could do:
```
ParseNode tree => LayoutNode tree => HtmlNode tree
```
where
```
LayoutNode =
   | Box
   | Rule
   | Kern
   | Glue
   | Glyph
```
The benefit of this would be being able to have different rendering backends so that we could do any of the following:
```
LayoutNode tree => HtmlNode tree
LayoutNode tree => SvgNode tree
LayoutNode tree => canvas render
```
Having a tree of `LayoutNode`s would allow us to do any post-processing such as bin cancellation on the `LayoutNode` tree once instead of having to have separate copies of the code for each of the final trees.